test/framework/src/main/java/org/elasticsearch/test/rest/yaml/ObjectPath.java

```diff
@@ -148,4 +149,20 @@ private Object evaluate(String key, Object object, Stash stash) throws IOExcepti
 
         return list.toArray(new String[list.size()]);
     }
+
+    /**
+     * Create a new {@link XContentBuilder} from the xContent object underlying this {@link ObjectPath}.
+     * This only works for {@link ObjectPath} instances created from an xContent object, not from nested
+     * substructures. We throw an {@link UnsupportedOperationException} in those cases.
+     */
+    @SuppressWarnings("unchecked")
+    public XContentBuilder toXContentBuilder(XContent xContent) throws IOException {
+        XContentBuilder builder = XContentBuilder.builder(xContent);
+        if (this.object instanceof Map) {
+            builder.map((Map<String, Object>) this.object);
+        } else {
+            throw new UnsupportedOperationException("Only ObjectPath created from a map supported.");
```

Comment:
I don't think we should to that here and now when we don't need it. ObjectPath is whacky already since it allows to wrap any arbitrary kind of Object, not only lists and maps (I don't think it does anything useful with other kinds of Object, but nethertheless...) I don't think "faking" it can produce any meaningful xContent if the underlying Object is not a Map<String, Object> stretches things a bit.